components/learn/modules/ROOT/pages/on-dapps.adoc

```diff
@@ -1,8 +1,160 @@
+
+function App() {
+  // Get web3 context via @openzeppelin/network/react hook
+  const web3Context = useWeb3(`wss://mainnet.infura.io/ws/v3/${PROJECT_ID}`);
+  const { lib: web3, networkId, accounts, providerName } = web3Context;
+
+  // Methods for requesting accounts access
+  const requestAuth = (web3Context) => web3Context.requestAuth();
+  const requestAccess = useCallback(() => requestAuth(web3Context), []);
+
+  // Querying account balance
+  const [balance, setBalance] = useState(0);
+  const getBalance = useCallback(async () => {
+    setBalance(
+      accounts && accounts.length > 0 
+      ? (await web3.eth.getBalance(accounts[0]))
+      : 'Unknown')
+  }, [accounts, web3.eth, web3.utils]);
+  useEffect(() => {
+    getBalance();
+  }, [accounts, getBalance, networkId]);
+
+  // Show all information to the user
+  return (
+    <div>
+      <div>Network: {networkId || 'No connection'}</div>
+      <div>Your address: {accounts ? accounts[0] : 'Unknown'}</div>
+      
+      {accounts && accounts.length ? (
+        <div>Accounts access granted</div>
+      ) : !!networkId && providerName !== 'infura' ? (
+        <button onClick={requestAccess}>Request Access</button>
+      ) : (
+        <div>No accounts access</div>
+      )}
+    </div>
+  );
+}
+```
+
+NOTE: You can easily xref:on-gsn.adoc[enable meta-transaction support] in your dapp when using the network library by http://example.com[adding a `gsn` option] to the React hook.
+
+You can follow a step-by-step tutorial on building a dapp from scratch using the network-js library in the http://example.com[Building a network powered dapp] guide.
+
+[[hot-loader]]
+== Hot-loading Contracts
+
+Now that we have a working connection to the Ethereum network, we can start interacting with the contracts we have coded and xref:deploy-and-interact.adoc[deployed using the CLI]. While we can create a https://web3js.readthedocs.io/en/v1.2.4/web3-eth-contract.html[regular `web3` contract instance] using the ABI and address, we can instead leverage xref:on-upgrades.adoc[OpenZeppelin upgrades] to hot-load our contracts.
+
+This means that, once we have deployed a contract and loaded it in our running React dapp, the app will **automatically pick up any changes to the Solidity code and upgrade the contract** during development. The https://github.com/OpenZeppelin/solidity-loader[hot loader] will watch all your Solidity contracts, and whenever you recompile, it will automatically upgrade the contract on the local development network - keeping the same address and state, and just changing the code to the new version you wrote.
+
+NOTE: The hot-loader is implemented as at https://webpack.js.org/[webpack] plugin. It will not work if you are using a different bundler for your application.
+
+To set up the hot-loader, you need to install `@openzeppelin/solidity-loader` and `json-loader`, and add the following https://webpack.js.org/configuration/module/#rule[rule] to your webpack configuration:
```

Comment:
I'd consider dropping this and focusing on showcasing how great the loader is, instead of explaining how to use it (which will be covered by its own documentation). This is the approach followed by the Network guide above, and it works great.

Reply:
Disagree. I think there is value in showing that it is _just_ another webpack plugin to set up, and there is no more magic involved.

Reply:
Ah, that might be me not being used to webpack then.